packages/relay/src/lib/subscriptionController.ts

```diff
@@ -0,0 +1,134 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// import { Poller } from './poller';
+import { Logger } from 'pino';
+import LRU from "lru-cache";
+import constants from "./constants";
+import crypto from "crypto";
+
+export interface Subscriber {
+    connection: any,
+    subscriptionId: string
+}
+
+const LOGGER_PREFIX = 'Subscriptions:';
+
+export class SubscriptionController {
+    // private poller: Poller;
+    private logger: Logger;
+    private subscriptions: {[key: string]: Subscriber[]};
+    private cache;
+
+    // constructor(poller: Poller, logger: Logger) {
+    constructor(logger: Logger) {
+        // this.poller = poller;
+        this.logger = logger;
+        this.subscriptions = {};
+        this.cache = new LRU({ max: constants.CACHE_MAX, ttl: 2000 });
+    }
+
+    createHash(data) {
+        return crypto.createHash('sha3').update(data.toString()).digest('hex');
+    }
+
+    // Generates a random 16 byte hex string
+    generateId() {
+        return "0x" + crypto.randomBytes(16).toString('hex');
+    }
+
+    subscribe(connection, event: string, filters?: {}) {
+
+        let tag: any = {event};
+        if (filters && Object.keys(filters).length) {
+            tag.filters = filters;
+        }
+
+        tag = JSON.stringify(tag);
```

Comment:
Q: are we concerned about the size of the tag string? Does it need to be shortened?

Reply:
Not really. The idea of this tag is to be unique to every event and filter combination and to provide an easy way to extract that data from it. We could create some more compressed format and optimize on some of the commas, quotation marks and curly brackets, but that would be ~5% of the total length at most, I think converting it to a simple json should be fine.